File: tests/consensus_test.go
package tests

import (
    "testing"
    "time"
    "stg-chain/consensus"
)

func TestIBFTConsensus_Start(t *testing.T) {
    validators := []string{"validator1", "validator2", "validator3"}
    engine := consensus.NewIBFTConsensus(validators, 30000, 3*time.Second)

    err := engine.Start()
    if err != nil {
        t.Errorf("Consensus failed to start: %v", err)
    }
}

func TestIBFTConsensus_ProposeValidateCommit(t *testing.T) {
    validators := []string{"validator1", "validator2", "validator3"}
    engine := consensus.NewIBFTConsensus(validators, 30000, 3*time.Second)

    blockData := []byte("test-block")
    err := engine.ProposeBlock(blockData)
    if err != nil {
        t.Errorf("Failed to propose block: %v", err)
    }

    if !engine.ValidateBlock(blockData) {
        t.Errorf("Block validation failed")
    }

    err = engine.CommitBlock(blockData)
    if err != nil {
        t.Errorf("Failed to commit block: %v", err)
    }
}

File: tests/consensus.test.ts
import { IBFTConsensus } from "../consensus/engine";

describe("IBFTConsensus", () => {
  const validators = ["validator1", "validator2", "validator3"];
  const engine = new IBFTConsensus(validators, 30000, 3);

  test("should start consensus engine", () => {
    expect(() => engine.start()).not.toThrow();
  });

  test("should propose, validate, and commit block", () => {
    const blockData = "test-block";

    expect(() => engine.proposeBlock(blockData)).not.toThrow();
    expect(engine.validateBlock(blockData)).toBe(true);
    expect(() => engine.commitBlock(blockData)).not.toThrow();
  });
});
